lmctl/drivers/lm/security.py

```diff
@@ -76,7 +79,14 @@ def __init__(self, auth_address, username=None, password=None, client_id=None, c
             if self.__username is not None:
                 # Using password auth
                 if self.__client_id is not None:
-                    client_builder.user_pass_auth(username=self.__username, password=self.__password, client_id=self.__client_id, client_secret=self.__client_secret)
+                    if self.__auth_mode == OKTA_MODE:
+                        client_builder.okta_user_pass_auth(username=self.__username, password=self.__password,
```

Comment:
What about `okta_client_credentials_auth`? That could be used? I think trying to combine the checks for Okta with the legacy Oauth has led to this confusion. 
Can we add `elif self.__auth_mode.lower() == OKTA_AUTH_MODE:` under the check for Zen/Token and handle Okta separate? Leaving the `else` block for `#Oauth` as it was?

Reply:
Addressed